evm/vm/forks/frontier/validation.py

```diff
@@ -18,7 +18,11 @@ def validate_frontier_transaction(vm_state, transaction):
     if sender_balance < total_cost:
         raise ValidationError("Sender account balance cannot afford txn")
 
-    if vm_state.block_header.gas_used + transaction.gas > vm_state.block_header.gas_limit:
+    if vm_state.receipts:
+        gas_used = vm_state.receipts[-1].gas_used
```

Comment:
What do you think about adding a`@propery` method for the `vm_state` class since this block of logic is repeated at least one more time elsewhere in the code.

```python
@propert
def gas_used(seelf):
    if vm_state.receipts:
        return ...
    else:
        return 0
```